docs/03-declarative-layout.md

```diff
@@ -424,3 +436,27 @@ Code:
 
     s.mywibox : setup (three_circle)
 
+### Instantiation rules
+
+Whenever it can, Awesome tries to be asynchronous. This can take various form
+depending on the situation. For example, the `connect_signal` method allows to
+execute code when an event arrives. `awful.screen.connect_for_each_screen` also
+allow to instantiate various elements when a new screen is added. In the later
+case, it is why some widgets are added as properties to other objects instead of
+being global variables like in previous versions of Awesome.
+
+However, there is a case where this isn't enough an another abstract widget has
+to be used. This concept is called the `widget_template` and is an optional
+property of many widgets such as the `awful.widget.taglist`,
+`awful.widget.tasklist` and `naughty.widget.box`. These templates are **table**
```

Comment:
Maybe I'm missing it, but where is `naughty.widget.box`? I only see `naughty.layout.box`.

Also, *are a **table**